src/lib/services/access-service.ts

```diff
@@ -65,6 +65,9 @@ export type IdPermissionRef = Pick<IPermission, 'id' | 'environment'>;
 export type NamePermissionRef = Pick<IPermission, 'name' | 'environment'>;
 export type PermissionRef = IdPermissionRef | NamePermissionRef;
 
+type APIUser = Pick<IUser, 'id' | 'permissions'> & { isAPI: true };
+type NonAPIUser = Pick<IUser, 'id'> & { isAPI?: false };
```

Comment:
This is unrelated but it's a QoL improvement. When using an API user this service reads the permissions from the user (they should be injected before this method is called), but when called with a non API user, we just ignore the permissions in the object and fetch them again from the DB